lib/templateFactory.js

```diff
@@ -146,56 +146,54 @@ class TemplateFactory extends BaseFactory {
         let isLatest = true;
         let isTrusted = false;
 
+        const [, configMajor, configMinor] = VERSION_REGEX.exec(config.version);
+
+        let latestTemplate;
+
         return super
             .list({ params: { namespace: result.namespace, name: result.name, latest: true } })
             .then(latestTemplates => {
-                const latestTemplate = latestTemplates[0];
-                const [, configMajor, configMinor] = VERSION_REGEX.exec(config.version);
+                latestTemplate = latestTemplates[0];
 
+                if (latestTemplate) {
```

Comment:
I fixed it because it is unnecessary as logic.
The number of times of accessing db will increase a little, although I figured that to simplify the logic is better.